lib/Common/Core/ConfigParser.cpp

```diff
@@ -337,19 +337,70 @@ void ConfigParser::ParseConfig(HANDLE hmod, CmdLineArgsParser &parser)
         {
             return;
         }
+
         if (_wfopen_s(&configFile, filename, _u("r, ccs=UNICODE")) != 0 || configFile == nullptr)
         {
             return;
         }
     }
+#else
+    // Two-pathed for a couple reasons
+    // 1. PAL doesn't like the ccs option passed in.
```

Comment:
Well, when I copy and paste the `wfopen` calls to non-Windows part, it works as is.

`MapFileOpenModes` doesn't handle `ccs=Unicode`, it simply ignores.

Looking at the 2 versions of your implementation, isn't the same thing you are trying to achieve ? What I'm trying to say is that, well, it's not needed to code twice.

```
    if (_wfopen_s(&configFile, filename, _u("r, ccs=UNICODE")) != 0 || configFile == nullptr)
VS
    if (_wfopen_s(&configFile, filename, _u("r")) != 0 || configFile == nullptr)
```